### PR TITLE
Created a mailer to handle the Adoption Application forms.

### PR DESCRIPTION
**Pull request by JessSumner**
Why:
When a form is completed and submitted from the website the rescue needs to be notified and have access to the data.

This PR:
Generates a mailer for the Adoption Application and includes a method to send an email.
Has added securely an admin email that can be called without making this information public.
Corrected issues with the radio buttons not actually saving an input to the database.
Has created an HTML and text template for the email.
Includes a preview of the HTML email.

### CODE REVIEW

**Comment by JessSumner** [comment]
@MattMSumner @halogenandtoast @kenyonj I'm less confident in this PR as I covered several new topics. If you have a few moments I would greatly appreciate your feedback. Thank you

File: app/mailers/adoption_application_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class AdoptionApplicationMailer < ApplicationMailer
+  def new_adoption_application(adoption_application)
```
Comment (halogenandtoast) [inline app/mailers/adoption_application_mailer.rb]: I think I would rename both the class and this method. It seems like this is more contextual than your class implies. For instance I would choose `NewAdoptionApplicationMailer` which describes better what it's purpose is (to send mails when a new application is created). Then I'd probably just call my method `notification`. What do you think?

File: app/views/adoption_application_mailer/_table_row.html.erb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<tr>
+  <td style="border:solid; border-width:1px; font-weight:bold;"><%= data_entry.to_s %></td>
```
Comment (halogenandtoast) [inline app/views/adoption_application_mailer/_table_row.html.erb]: `to_s` isn't necessary here as <%= thing %> calls `to_s` on `thing` automatically.

File: app/views/adoption_application_mailer/_text_table_row.text.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<%= data_entry.to_s %>: <%= adoption_application.public_send(data_entry) %>
```
Comment (halogenandtoast) [inline app/views/adoption_application_mailer/_text_table_row.text.erb]: `to_s` isn't needed here either.

File: app/views/adoption_application_mailer/new_adoption_application.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />
+  </head>
+  <body>
+    <h1 style="color:#363691">A new adoption application form has been submitted.</h1>
+    <div>
+      <%= render "adoption_application_mailer_content",
```
Comment (halogenandtoast) [inline app/views/adoption_application_mailer/new_adoption_application.html.erb]: I don't think a partial was necessary here.

File: app/views/adoption_application_mailer/new_adoption_application.text.erb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+A new adoption application form has been submitted.
+
+<%= render "text_adoption_application_mailer_content",
```
Comment (halogenandtoast) [inline app/views/adoption_application_mailer/new_adoption_application.text.erb]: I don't think a partial was necessary here.

File: app/views/adoption_applications/_form_fields.html.erb [hunk 6]
```diff
@@ -23,14 +23,17 @@
   <%= f.text_field :email %>
   
   <div class="radio"> 
-    <%= radio_button_tag(:current_rats, "yes") %>
-    <%= label_tag(:current_rats_yes, "I currently own one or more rats") %></br>
+    <%= f.radio_button :current_rats, true %>
+    <%= f.label :current_rats, "I currently own one or more rats", :value => "true" %></br>
```
Comment (halogenandtoast) [inline app/views/adoption_applications/_form_fields.html.erb]: I'd suggest using the newer hash syntax

``` erb
<%= f.label :current_rats, "I currently own one or more rats", value: "true" %>
```

File: app/views/adoption_applications/_form_fields.html.erb [hunk 7]
```diff
@@ -23,14 +23,17 @@
...
 
-    <%= radio_button_tag(:current_rats, "no", true) %>
-    <%= label_tag(:current_rats_no, "I do not currently own any rats") %>
+    <%= f.radio_button :current_rats, false, :checked => true %>
```
Comment (halogenandtoast) [inline app/views/adoption_applications/_form_fields.html.erb]: Same here with the hash syntax.

File: app/views/adoption_applications/_form_fields.html.erb [hunk 8]
```diff
@@ -23,14 +23,17 @@
...
+    <%= f.label :current_rats, "I do not currently own any rats",
+                :value => "false" %>
```
Comment (halogenandtoast) [inline app/views/adoption_applications/_form_fields.html.erb]: Same here with the hash syntax.

File: app/views/adoption_applications/_form_fields.html.erb (hunk 7 above)
Comment (halogenandtoast) [inline app/views/adoption_applications/_form_fields.html.erb]: I've never used the `checked` option before. Pretty neat!

File: app/views/adoption_applications/_form_fields.html.erb (hunk 6 above)
Comment (halogenandtoast) [inline app/views/adoption_applications/_form_fields.html.erb]: I've never used the `value` option before. Pretty neat!

File: app/views/adoption_application_mailer/_adoption_application_mailer_content.html.erb [hunk 9]
```diff
@@ -0,0 +1,50 @@
+<table style="border-collapse: collapse;">
+  <%= render "table_row", adoption_application: adoption_application,
```
Comment (halogenandtoast) [inline app/views/adoption_application_mailer/_adoption_application_mailer_content.html.erb]: Not sure how I feel about this partial usage. It's still a lot of repetition. You could do something like.

``` ruby
class AdoptionApplication
  PUBLIC_FIELDS =  %i(name address city cell state zip email current_rates
      current_rat_count current_rat_gender current_rat_age previous rates
      previous_rate_count previous_rat_lifespan cause_of_death vet_care
      vet_name  cage_description cage_location household_members
      children_ages additional_animals interactions_time preferred_characteristics)
```

then here maybe

``` erb
<% AdoptionApplication::PUBLIC_FIELDS.each do |field| %>
  <tr>
    <td style="border:solid; border-width:1px; font-weight:bold;"><%= field %></td>
    <td style="border:solid; border-width:1px; width: auto;"><%= @adoption_application.attributes[field] %></td>
  </tr>
<% end %>
```

Comment (halogenandtoast) [inline app/views/adoption_application_mailer/_adoption_application_mailer_content.html.erb]: That avoids the public send, and can be reused in the text version as well.

Comment (JessSumner) [inline app/views/adoption_application_mailer/_adoption_application_mailer_content.html.erb]: @halogenandtoast This suggestion is exactly what I hoped was possible but couldn't see how to do. I've changed it slightly but thank you so much for the guidance.

**Pull request merged**